__tests__/login.js

```diff
@@ -14,82 +14,140 @@ it('renders the login page at /login', async () => {
+  fireEvent.change(passwordInput, { target: { value: 'password' } })
+
+  // And click submit
+  const button = getByText('Log in');
+  fireEvent.click(button);
+
+  // Then I should be logged in and see the home page with no orgs or clusters.
+  await wait(() => {
+    // Verify we are now on the layout page and I can see my username
+    expect(getByText(/developer@giantswarm.io/i)).toBeInTheDocument();
+    expect(getByText(/Welcome to Giant Swarm!/i)).toBeInTheDocument();
+    expect(getByText(/There are no organizations yet in your installation./i)).toBeInTheDocument();
+  });
+
+  // Assert that the mocked responses got called, tell them to stop waiting for
+  // a request.
+  authTokensRequest.done();
+  userInfoRequest.done();
+  infoRequest.done();
+  orgRequest.done();
+  clustersRequest.done();
+  appcatalogsRequest.done();
 });
 
-it.skip('tells the user to give a password if they leave it blank', async () => {
-  // To be implemented
+it('tells the user to give a password if they leave it blank', async () => {
+  // Given I arrive at the login page with nothing in the state.
+  const state = {};
+  const div = document.createElement('div');
+  const { getByText, getByLabelText } = renderRouteWithStore('/login', div, state);
+
+  // When I type in my email but not my password.
+  const emailInput = getByLabelText('Email');
+  fireEvent.change(emailInput, { target: { value: 'developer@giantswarm.io' } })
+
+  // And click submit
+  const button = getByText('Log in');
+  fireEvent.click(button);
+
+  // Then I should see a message reminding me not to leave my password blank.
+  await wait(() => {
+    expect(getByText(/Please enter your password./i)).toBeInTheDocument();
+  });
 });
 
-it.skip('tells the user to give a email if they leave it blank', async () => {
-  // To be implemented
+it('tells the user to give a email if they leave it blank', async () => {
+  // Given I arrive at the login page with nothing in the state.
+  const state = {};
+  const div = document.createElement('div');
+  const { getByText, getByLabelText } = renderRouteWithStore('/login', div, state);
+
+  // When I type in my password but not my email.
+  const passwordInput = getByLabelText('Password');
+  fireEvent.change(passwordInput, { target: { value: 'password' } })
+
+  // And click submit
+  const button = getByText('Log in');
+  fireEvent.click(button);
+
+  // Then I should see a message reminding me not to leave my password blank.
+  await wait(() => {
+    expect(getByText(/Please provide the email/i)).toBeInTheDocument();
+  });
 });
 
-it.skip('shows an error if the user logs in with invalid credentials', async () => {
-  // To be implemented
+it('shows an error if the user logs in with invalid credentials', async () => {
+  // Given I have a Giant Swarm API that does not accept my login attempt
+
+  // The failed 401 response to the login call
+  const authTokensRequest = nock('http://localhost:8000')
+  .post('/v4/auth-tokens/')
+  .reply(401);
```

Comment:
this causes the test to log the error, (since I have these `console.error` calls in the catch blocks of most of the actions. But that is expected. It might be confusing though in the test output.